pkg/leclient/utils.go

```diff
@@ -36,3 +41,37 @@ func GetSecret(kubeClient client.Client, secretName, namespace string) (*corev1.
 
 	return s, nil
 }
+
+// ExponentialBackOff will sleep for a specific amount of time, determined by the number of API failures
+// which are recorded in the CertificateRequestConditions under Status.
+func ExponentialBackOff(cr *certman.CertificateRequest, queryType string) (nil, error) {
+	for i, condition := range cr.Status.Conditions {
+		if string(condition.Type) == queryType {
+			failCount, err := strconv.Atoi(string(condition.Status))
+			if err != nil {
+				return err
+			}
+			// Sleep an exponential number of seconds after each API call failure.
+			// For example: 2 seconds, 4 seconds, 16 seconds, 32 seconds...
+			seconds := 200 * time.Millisecond
+			sleeptime := seconds << uint(failCount)
```

Comment:
Let's also put a cap in it. Something like 
`sleeptime = math.Min(100, sleeptime)`
or whatever number we choose for a max backoff length.